Libraries/process/Tests/test_DefaultLauncher.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <gtest/gtest.h>
+#include <process/DefaultLauncher.h>
+#include <process/LaunchResult.h>
+#include <process/MemoryContext.h>
+#include <libutil/DefaultFilesystem.h>
+
+#include <unistd.h>
+
+using process::DefaultLauncher;
+using process::MemoryContext;
+using libutil::DefaultFilesystem;
+
+TEST(DefaultLauncher, CapturesStdout)
+{
+    DefaultLauncher launcher;
+    auto filesystem = DefaultFilesystem::GetDefaultUNSAFE();
+
+    std::vector<std::string> arguments;
+    arguments.push_back("foo");
+    std::unordered_map<std::string, std::string> environment;
+    auto echo = MemoryContext(
+        "/bin/echo",
+        "/",
+        arguments,
+        std::unordered_map<std::string, std::string>(),
+        ::getuid(),
+        ::getgid(),
```

Comment:
`0` might work when using a `MemoryFilesystem`. Using a `DefaultFilesystem` I get `setuid: Operation not permitted` and `setgid: Operation not permitted`. I'll definitely try this if I can figure out a way to use `MemoryFilesystem`, though!